spring-boot-admin-server/src/main/java/de/codecentric/boot/admin/registry/HealthStatusUpdater.java

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright 2013-2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package de.codecentric.boot.admin.registry;
+
+import java.lang.invoke.MethodHandles;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Map.Entry;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.boot.actuate.health.Health;
+import org.springframework.boot.actuate.health.Status;
+import org.springframework.context.ApplicationEventPublisher;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.client.RestTemplate;
+
+import de.codecentric.boot.admin.event.ClientApplicationHealthStatusChangedEvent;
+import de.codecentric.boot.admin.model.Application;
+import de.codecentric.boot.admin.model.StatusInfo;
+import de.codecentric.boot.admin.registry.store.ApplicationStore;
+
+/**
+ * this {@link StatusUpdater} can be activated by defining a bean such as
+ * <pre class="code">
+ *	&#064;Primary
+ *	&#064;Bean(name={"statusUpdater", "healthStatusUpdater"})
+ *	public HealthStatusUpdater statusUpdater() {
+ *		RestTemplateBuilder builder = restTemplBuilder
+ *				.messageConverters(new MappingJackson2HttpMessageConverter())
+ *				.errorHandler(new DefaultResponseErrorHandler() {
+ *			&#064;Override
+ *			protected boolean hasError(HttpStatus statusCode) {
+ *				return false;
+ *			}
+ *		});
+ *		if (managementServerProperties.getSecurity().isEnabled() && securityProperties.getBasic().isEnabled()) {
+ *			builder = builder.basicAuthorization(securityProperties.getUser().getName(), 
+ *				securityProperties.getUser().getPassword());
+ *		}
+ *		HealthStatusUpdater statusUpdater = new HealthStatusUpdater(builder.build(), applicationStore);
+ *		statusUpdater.setStatusLifetime(adminServerProperties.getMonitor().getStatusLifetime());
+ *		return statusUpdater;
+ *	}
+ * </code>
+ * 
+ * @author Johannes Edmeier
+ * @author dbs
+ * @since 1.4.4
+ * @version 1.0.0
+ *
+ */
+public class HealthStatusUpdater extends StatusUpdater {
```

Comment:
Instead of writing a new StatusUpdater implementation I'd suggest to modify the existing one just setting the new `details` property of the `ClientApplicationStatusChangedEvent` to either the caught exception or the body of the response.